kong/db/dao/certificates.lua

```diff
@@ -99,8 +108,26 @@ function _Certificates:update(cert_pk, cert, options)
     end
   end
 
+  local snis
+  snis, err, err_t = self.db.snis:list_for_certificate(cert_pk)
+  if not snis then
+    return nil, err, err_t
+  end
+
   -- update certificate if necessary
   if cert.key or cert.cert then
+    if cert.key == null then
+      -- user tries to remove private key from cert
+      if (name_list and #name_list ~= 0) or #snis ~= 0 then
+        local err_t = self.errors:schema_violation({
+          key = "one or more SNI names are still associated with this" ..
+                 " certificate, private key cannot be removed",
```

Comment:
In the same vein:
```
key = "cannot be removed, since one or more SNIs are still associated to this Certificate"
```

(Since `key` already is the field name in the error object, this is closer to our other schema errors for individual fields)